### ISSUE TITLE
Using Agent on blade

### ISSUE THREAD

**Issue by leandroluk** [jenssegers/agent#64]
Hi, i am new to laravel and I wonder if there is any way I can use the Agent directly on my blade file so that according to him I can change the type of content I'll use (so is better than having to use browserhack )

sorry for my english, google help me =]

**Comment by ahmedash95** [comment]
Hi @leandroluk , I guess this should work perfect with you 
```
@if((new Jenssegers\Agent\Agent)->isAndroidOS)
<h1> From Android </h1>
@elseif((new Jenssegers\Agent\Agent)->isSafari)
<h1> From Safari</h1>
@endif
```

**Comment by timgavin** [comment]
@leandroluk [Here's an example](http://stackoverflow.com/a/39800742/2101328) for using it globally in blade templates with a Service Provider.

**Comment by leandroluk** [comment]
Thx my friend!

Em 01/10/2016 11:43, "tim gavin" <notifications@github.com> escreveu:

**Comment by TonnyORG** [comment]
@leandroluk @hmedash95 @timgavin you guys can also follow *the-Laravel-way* and use the Facade (looks more readable):

```php
@if (Agent::isAndroidOS() )
<h1> From Android </h1>
@elseif (Agent::isSafari())
<h1> From Safari</h1>
@endif
```

**Comment by ahmedash95** [comment]
Don't know it has a facade class , nice hint @TonnyORG

**Comment by emergingdzns** [comment]
I'm having an issue using this tool. On my local machine (running MAMP) I have a blade that looks like this:
```
@if (Agent::isMobile())
    @include('mobile.dashboard')
@else
    @include('desktop.dashboard')
@endif
```
(I also tried `@if (Jenssegers\Agent\Agent::isMobile())`)

Doing it this way, I get this error: `Non-static method Mobile_Detect::isMobile() should not be called statically`

I changed it to this:
```
<?
use Jenssegers\Agent\Agent;

$agent = new Agent();
?>
@if ($agent->isMobile())
    @include('mobile.dashboard')
@else
    @include('desktop.dashboard')
@endif
```

And then it works on my local system. BUT, when I upload to our Amazon server, I am getting an error that `$agent` is not a valid variable.

Is there a configuration necessary or a PHP package that has to be compiled into it?

Could really use some help.

Thanks!

**Comment by TonnyORG** [comment]
@emergingdzns your config/app.php already has the respective alias to make it works as facade?

For more information read this:
https://github.com/TonnyORG/agent/blob/33043abec5a61d5ae6db2677c2c6457690801c71/README.md#laravel-optional (Laravel (optional))

**Comment by emergingdzns** [comment]
Yes. I have the `'Agent' => Jenssegers\Agent\Facades\Agent::class,` line in my app.php file.

I just can't seem to make it work. It's weird to me that in the blade, I have the use statement and then instantiate the `$agent = new Agent();` but then the blade says that $agent isn't a valid variable... Makes no sense.

**Comment by ahmedash95** [comment]
@emergingdzns try to use this in blade and tell me if it works for you 
```blade 
@if (\Jenssegers\Agent\Facades\Agent::isMobile())
    @include('mobile.dashboard')
@else
    @include('desktop.dashboard')
@endif
```
just to make sure if it reads the Agent facade path right or not

**Comment by DanielUche** [comment]
Ihave tried it countless times in laravel 5.3 and it does not work using Facades

**Comment by DanielUche** [comment]
My solution was to instantiate the agent class everytime I needed it, which is not the laravel way

**Comment by dericcain** [comment]
I'm using Laravel 5.3 and it is working for me in the controller. Maybe the declaration is wrong?

```php
// is this the declaration being used? This works for me.
use Jenssegers\Agent\Facades\Agent;

// use the facade as normal
'isTablet' => Agent::isTablet()
```

Looks like I am using v2.4 of the Agent

**Comment by ludo237** [comment]
Why just don't user an helper function 😕
